### ISSUE TITLE
WebAPI calling WCF service with exception: "An asynchronous module or handler completed while an asynchronous operation was still pending"

### ISSUE THREAD

**Issue by henkish** [dotnet/wcf#2725]
I have an issue where it seems there is some kind of race condition.

The setup is a .net 462 web api service (running Microsoft.AspNet.WebApi version 5.2.3) has a async method, with signature looking like: `public async Task<HttpResponseMessage> Get(string id);`

In the method I make a WCF method call using connected service running System.ServiceModel latest preview version (4.5.0-preview1-26228-01 - but issue also in latest stable version 4.4.1). When the WCF service is executed without exception - everything works fine. But when WCF service is executed with exception (catched and handled by my code) the controller action returns 500 exception with error message: "An asynchronous module or handler completed while an asynchronous operation was still pending"

So basically:

```c#
public async Task<HttpResponseMessage> Get(string id)
{
    try
    {
        var client = CreateWcfClient();
        var result = await client.MethodAsync(id);
        return Request.CreateResponse(HttpStatusCode.OK, result);
    }
    catch (Exception e)
    {
        return Request.CreateResponse(HttpStatusCode.OK, new string[0]);
    }
}
```

And if catch is triggered, I get a 500 error with message: "An asynchronous module or handler completed while an asynchronous operation was still pending".

Anyone experiencing the same issue?
Am I doing something wrong?

**Comment by Lxiamail** [comment]
Looks like you need to call client.Abort() in catch statement.

**Comment by KKhurin** [comment]
Hey @henkish, thanks for reporting this. 
This is likely due to something kicking off an async void operation in case of an error (which ASP.NET sync context doesn't like). 

What binding do you use? We'll try to repro it locally. There are also ways to debug it in your repro. Please let us know.

**Comment by Lxiamail** [comment]
@henkish Please let us know if you have any more question.
